scite/papers.py

```diff
@@ -57,5 +61,9 @@ def get_papers(dois: DOIs) -> JSON:
     -----
     Docs: https://api.scite.ai/docs#operation/get_papers_papers_get
     """
+    if not isinstance(dois, DOIs):
+        items = DOIs(ids=[{"id": item} for item in dois])
+        dois = {item.id for item in items.ids}

```

Comment:
That's what I mean when I say that the code becomes harder to follow. Maybe I should add some comments or rewrite it altogether